### ISSUE TITLE
Question about the reset_state Function Called in the reset() Method of eval_agent/envs/alfworld_env.py

### ISSUE THREAD

**Issue by observerz** [WangHanLinHenry/SPA-RL-Agent#5]
Hi,

   I hope you're doing well. I have a question regarding the reset_state function that is called within the reset() method of the eval_agent/envs/alfworld_env.py(line 94) file. I'm having trouble understanding what this function does and how it contributes to the overall functionality of the environment.

Could you please provide some clarification on its purpose and how it works? I would really appreciate your help!

Thank you!

**Comment by WangHanLinHenry** [comment]
Hello, @observerz 

Here is the illustration regarding how reset() method works:
Each piece of data corresponds to a unique initial state of the environment. After executing a series of actions, the simulator evaluates the environment state to determine whether the tasks were successfully completed. Therefore, to ensure fairness in each evaluation, it is essential to initialize the environment state at the beginning. And the reset() function could initialize the environment state.

**Comment by observerz** [comment]
Oh, it seems that I did not clearly describe my question. I would like to ask about the specific function of the code **self.env.reset_states(game_files)** on **line 94** in the file **eval_agent/envs/alfworld_env.py**. This function does not appear to exist in any version of TextWorld. Could you please explain it? Thank you!

**Comment by WangHanLinHenry** [comment]
Hi, this function was not written by me. It is included directly in the file eval_agent/envs/alfworld_env.py after following the installation instructions.

Below is the version configuration of the library I run in the alfworld environment. You can refer to it for reference.

Package                           Version
--------------------------------- -------------
accelerate                        0.34.2
ai2thor                           2.1.0
aiohappyeyeballs                  2.4.3
aiohttp                           3.10.10
aiohttp-cors                      0.8.0
aiosignal                         1.3.1
airportsdata                      20250224
alfworld                          0.3.4
annotated-types                   0.7.0
anyio                             4.6.2.post1
astor                             0.8.1
asttokens                         2.4.1
async-timeout                     4.0.3
attrs                             24.2.0
backoff                           2.2.1
bcrypt                            4.2.0
beautifulsoup4                    4.11.1
blake3                            1.0.4
blis                              0.7.11
cachetools                        5.5.2
catalogue                         2.0.10
certifi                           2024.8.30
cffi                              1.17.1
charset-normalizer                3.4.0
cleantext                         1.1.4
click                             8.1.7
cloudpathlib                      0.20.0
cloudpickle                       3.1.0
cmake                             3.30.5
colorama                          0.4.6
coloredlogs                       15.0.1
colorful                          0.5.6
comm                              0.2.2
compressed-tensors                0.8.0
confection                        0.1.5
contourpy                         1.3.0
cryptography                      43.0.3
cycler                            0.12.1
cymem                             2.0.8
Cython                            3.0.11
datasets                          2.19.1
debugpy                           1.8.7
decorator                         5.1.1
deepspeed                         0.16.5
depyf                             0.18.0
dill                              0.3.6
diskcache                         5.6.3
distlib                           0.3.9
distro                            1.9.0
docker                            6.1.3
docker-compose                    1.29.2
docker-pycreds                    0.4.0
dockerpty                         0.4.1
docopt                            0.6.2
docstring_parser                  0.16
einops                            0.8.0
en-core-web-lg                    3.6.0
env                               0.1.0
eval_type_backport                0.2.0
exceptiongroup                    1.2.2
executing                         2.1.0
faiss-cpu                         1.7.4
Farama-Notifications              0.0.4
fast_downward_textworld           20.6.2
fastapi                           0.115.3
filelock                          3.16.1
flash_attn                        2.7.4.post1
Flask                             2.1.2
flatbuffers                       24.3.25
fonttools                         4.54.1
frozenlist                        1.5.0
fsspec                            2024.3.1
gdown                             5.2.0
gguf                              0.10.0
gitdb                             4.0.11
GitPython                         3.1.43
google-api-core                   2.24.2
google-auth                       2.38.0
googleapis-common-protos          1.69.2
grpcio                            1.71.0
gym                               0.24.0
gym-notices                       0.0.8
gymnasium                         1.0.0
h11                               0.14.0
hashids                           1.3.1
hjson                             3.1.0
httpcore                          1.0.6
httptools                         0.6.4
httpx                             0.27.2
huggingface-hub                   0.26.1
humanfriendly                     10.0
idna                              3.10
importlib_metadata                8.5.0
importlib_resources               6.4.5
iniconfig                         2.0.0
intercode-bench                   0.1.22
interegular                       0.3.3
ipdb                              0.13.13
ipykernel                         6.29.5
ipython                           8.18.1
itsdangerous                      2.2.0
jedi                              0.19.1
jericho                           3.3.0
Jinja2                            3.1.4
jiter                             0.9.0
joblib                            1.4.2
jsonschema                        4.23.0
jsonschema-specifications         2024.10.1
jupyter_client                    8.6.3
jupyter_core                      5.7.2
kiwisolver                        1.4.7
langcodes                         3.4.1
language_data                     1.2.0
lark                              1.2.2
latex2mathml                      3.77.0
lightgbm                          4.5.0
linkify-it-py                     2.0.3
llvmlite                          0.43.0
lm-format-enforcer                0.10.11
marisa-trie                       1.2.1
markdown-it-py                    3.0.0
markdown2                         2.5.1
MarkupSafe                        3.0.2
matplotlib                        3.9.2
matplotlib-inline                 0.1.7
mdit-py-plugins                   0.4.2
mdurl                             0.1.2
mementos                          1.3.1
memray                            1.16.0
mistral_common                    1.5.4
more-itertools                    10.5.0
mpi4py                            3.1.4
mpmath                            1.3.0
msgpack                           1.1.0
msgspec                           0.19.0
multidict                         6.1.0
multiprocess                      0.70.14
murmurhash                        1.0.10
mysql-connector-python            9.1.0
nanobind                          2.5.0
nest_asyncio                      1.6.0
networkx                          3.2.1
nh3                               0.2.18
ninja                             1.11.1.1
nltk                              3.9.1
nmslib                            2.1.1
numba                             0.60.0
numpy                             1.26.3
nvidia-cublas-cu11                11.11.3.6
nvidia-cublas-cu12                12.1.3.1
nvidia-cuda-cupti-cu11            11.8.87
nvidia-cuda-cupti-cu12            12.1.105
nvidia-cuda-nvrtc-cu11            11.8.89
nvidia-cuda-nvrtc-cu12            12.1.105
nvidia-cuda-runtime-cu11          11.8.89
nvidia-cuda-runtime-cu12          12.1.105
nvidia-cudnn-cu11                 9.1.0.70
nvidia-cudnn-cu12                 9.1.0.70
nvidia-cufft-cu11                 10.9.0.58
nvidia-cufft-cu12                 11.0.2.54
nvidia-curand-cu11                10.3.0.86
nvidia-curand-cu12                10.3.2.106
nvidia-cusolver-cu11              11.4.1.48
nvidia-cusolver-cu12              11.4.5.107
nvidia-cusparse-cu11              11.7.5.86
nvidia-cusparse-cu12              12.1.0.106
nvidia-ml-py                      12.570.86
nvidia-nccl-cu11                  2.21.5
nvidia-nccl-cu12                  2.21.5
nvidia-nvjitlink-cu12             12.4.127
nvidia-nvtx-cu11                  11.8.86
nvidia-nvtx-cu12                  12.1.105
onnxruntime                       1.19.2
openai                            1.68.2
opencensus                        0.11.4
opencensus-context                0.1.3
opencv-python                     4.9.0.80
opencv-python-headless            4.11.0.86
outcome                           1.3.0.post0
outlines                          0.0.46
outlines_core                     0.1.26
packaging                         24.1
pandas                            2.2.3
paramiko                          3.5.0
parse                             1.20.2
parso                             0.8.4
partial-json-parser               0.2.1.1.post5
pathlib_abc                       0.1.1
pathy                             0.11.0
peft                              0.13.2
pexpect                           4.9.0
pickleshare                       0.7.5
pillow                            11.1.0
pip                               24.2
platformdirs                      4.3.6
pluggy                            1.5.0
preshed                           3.0.9
progressbar2                      4.5.0
prometheus_client                 0.21.0
prometheus-fastapi-instrumentator 7.1.0
prompt_toolkit                    3.0.48
propcache                         0.2.0
proto-plus                        1.26.1
protobuf                          5.28.3
psutil                            6.1.0
ptyprocess                        0.7.0
pure_eval                         0.2.3
py-cpuinfo                        9.0.0
py-spy                            0.4.0
py4j                              0.10.9.9
pyairports                        2.1.1
pyarrow                           17.0.0
pyarrow-hotfix                    0.6
pyasn1                            0.6.1
pyasn1_modules                    0.4.1
pybind11                          2.6.1
pycountry                         24.6.1
pycparser                         2.22
pydantic                          2.10.6
pydantic_core                     2.27.2
Pygments                          2.18.0
pyjnius                           1.6.1
PyNaCl                            1.5.0
pynvml                            11.5.0
pyOpenSSL                         24.2.1
pyparsing                         3.2.0
pyrsistent                        0.20.0
pyserini                          0.17.0
PySocks                           1.7.1
pytest                            8.3.3
python-dateutil                   2.9.0
python-dotenv                     0.21.1
python-utils                      3.8.2
pytz                              2024.2
PyYAML                            5.4.1
pyzmq                             26.2.0
rank-bm25                         0.2.2
RapidFuzz                         3.10.1
ray                               2.38.0
referencing                       0.35.1
regex                             2024.9.11
requests                          2.31.0
requests-mock                     1.12.1
responses                         0.18.0
rich                              13.9.3
rpds-py                           0.20.0
rsa                               4.9
safetensors                       0.4.5
scienceworld                      1.1.3
scikit-learn                      1.2.2
scipy                             1.13.1
seaborn                           0.13.2
selenium                          4.2.0
sentence-transformers             3.2.1
sentencepiece                     0.2.0
sentry-sdk                        2.17.0
setproctitle                      1.3.3
setuptools                        75.1.0
shellingham                       1.5.4
shortuuid                         1.0.13
shtab                             1.7.1
six                               1.16.0
smart-open                        6.4.0
smmap                             5.0.1
sniffio                           1.3.1
sortedcontainers                  2.4.0
soupsieve                         2.6
spacy                             3.6.1
spacy-legacy                      3.0.12
spacy-loggers                     1.0.5
srsly                             2.4.8
stack-data                        0.6.2
starlette                         0.41.0
svgwrite                          1.4.3
sympy                             1.13.1
TatSu                             5.8.3
termcolor                         2.5.0
texttable                         1.7.0
textual                           2.1.2
textworld                         1.6.1
thefuzz                           0.20.0
thinc                             8.1.12
threadpoolctl                     3.5.0
tiktoken                          0.9.0
tokenizers                        0.21.1
tomli                             2.0.2
torch                             2.5.1+cu121
torchaudio                        2.5.1+cu121
torchvision                       0.20.1+cu121
tornado                           6.4.1
tqdm                              4.66.5
traitlets                         5.14.3
transformers                      4.50.1
trio                              0.27.0
trio-websocket                    0.11.1
triton                            3.1.0
trl                               0.7.7
typer                             0.9.4
typing_extensions                 4.12.2
tyro                              0.8.14
tzdata                            2024.2
uc-micro-py                       1.0.3
urllib3                           1.26.20
urllib3-secure-extra              0.1.0
uvicorn                           0.32.0
uvloop                            0.21.0
virtualenv                        20.29.3
vllm                              0.6.4.post1
wandb                             0.18.5
wasabi                            1.1.3
watchfiles                        0.24.0
wavedrom                          2.0.3.post3
wcwidth                           0.2.13
weasel                            0.4.1
websocket-client                  0.59.0
websockets                        13.1
Werkzeug                          2.0.3
wheel                             0.44.0
wrapt                             1.16.0
wsproto                           1.2.0
xformers                          0.0.28.post3
xgrammar                          0.1.17
xxhash                            3.5.0
yarl                              1.16.0
zipp                              3.20.2
